decoimpact/business/entities/rules/rolling_statistics_rule.py

```diff
@@ -100,14 +78,15 @@ def execute(self, value_array: _xr.DataArray, logger: ILogger) -> _xr.DataArray:
             DataArray: Aggregated values
         """
 
-        time_scale = get_dict_element(self._time_scale, self._time_scale_mapping)
+        time_scale = get_dict_element(
+            self.settings.time_scale, self.settings.time_scale_mapping
+        )
 
         time_dim_name = get_time_dimension_name(value_array, logger)
 
         result = self._perform_operation(
             value_array,
             time_dim_name,
```

Comment:
Why is the period removed?

Reply:
It is a property of this class so it does not need to be passed along to the _perform_operation method because it can access it itself (using the self property)